aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/BaseHandlerStd.java

```diff
@@ -497,7 +474,58 @@ protected boolean isDbInstanceStabilized(
             final ResourceModel model
     ) {
         final DBInstance dbInstance = fetchDBInstance(rdsProxyClient, model);
-        return DBInstanceStatus.Available.equalsString(dbInstance.dbInstanceStatus());
+        return DBInstanceStatus.Available.equalsString(dbInstance.dbInstanceStatus()) &&
+                isReplicationComplete(dbInstance) &&
+                isDBParameterGroupSyncComplete(dbInstance) &&
+                isNoPendingChanges(dbInstance) &&
+                isVpcSecurityGroupsActive(dbInstance) &&
+                isDomainMembershipsJoined(dbInstance) &&
+                isPromotionTierUpdated(dbInstance, model);
+    }
+
+    boolean isPromotionTierUpdated(final DBInstance dbInstance, final ResourceModel model) {
+        return dbInstance.promotionTier() == null || dbInstance.promotionTier().equals(model.getPromotionTier());
+    }
+
+    boolean isDomainMembershipsJoined(final DBInstance dbInstance) {
+        return Optional.ofNullable(dbInstance.domainMemberships()).orElse(Collections.emptyList())
+                .stream()
+                .allMatch(membership -> DOMAIN_MEMBERSHIP_JOINED.equals(membership.status()) ||
+                        DOMAIN_MEMBERSHIP_KERBEROS_ENABLED.equals(membership.status()));
+    }
+
+    boolean isVpcSecurityGroupsActive(final DBInstance dbInstance) {
+        return Optional.ofNullable(dbInstance.vpcSecurityGroups()).orElse(Collections.emptyList())
+                .stream()
+                .allMatch(group -> VPC_SECURITY_GROUP_STATUS_ACTIVE.equals(group.status()));
+    }
+
+    boolean isNoPendingChanges(final DBInstance dbInstance) {
+        final PendingModifiedValues pending = dbInstance.pendingModifiedValues();
+        return (pending == null) || (pending.dbInstanceClass() == null &&
+                pending.allocatedStorage() == null &&
+                pending.caCertificateIdentifier() == null &&
+                pending.masterUserPassword() == null &&
+                pending.port() == null &&
+                pending.backupRetentionPeriod() == null &&
+                pending.multiAZ() == null &&
+                pending.engineVersion() == null &&
+                pending.iops() == null &&
+                pending.dbInstanceIdentifier() == null
```

Comment:
What about remaining features? Should we add them?
```
pending.dbInstanceClass()
pending.licenseModel()
pending.storageType()
pending.caCertificateIdentifier()
pending.dbSubnetGroupName()
pending.pendingCloudwatchLogsExports()
pending.hasProcessorFeatures() ? this.processorFeatures() : null
pending.iamDatabaseAuthenticationEnabled()
pending.automationModeAsString()
pending.resumeFullAutomationModeTime()
```

Reply:
This set was takes from the legacy implementation. It won't hurt to add them, I guess. On it.